frontend/packages/kubevirt-plugin/src/k8s/requests/vm/create/common.ts

```diff
@@ -34,13 +43,37 @@ export const initializeCommonMetadata = (
 
   entity.addLabel(`${TEMPLATE_OS_LABEL}/${osID}`, 'true');
   entity.addLabel(`${TEMPLATE_FLAVOR_LABEL}/${settings[VMSettingsField.FLAVOR]}`, 'true');
-  entity.addLabel(
-    `${TEMPLATE_WORKLOAD_LABEL}/${settings[VMSettingsField.WORKLOAD_PROFILE]}`,
-    'true',
-  );
 
-  entity.addLabel(LABEL_USED_TEMPLATE_NAME, getName(template));
-  entity.addLabel(LABEL_USED_TEMPLATE_NAMESPACE, getNamespace(template));
+  if (settings[VMSettingsField.WORKLOAD_PROFILE]) {
+    entity.addLabel(
+      `${TEMPLATE_WORKLOAD_LABEL}/${settings[VMSettingsField.WORKLOAD_PROFILE]}`,
+      'true',
+    );
+  }
+
+  if (template) {
+    entity.addLabel(LABEL_USED_TEMPLATE_NAME, getName(template));
+    entity.addLabel(LABEL_USED_TEMPLATE_NAMESPACE, getNamespace(template));
+  }
 
   return entity;
 };
+
+export const initializeCommonVMMetadata = (
+  { vmSettings }: CreateVMEnhancedParams,
+  entity: MutableVMWrapper,
+) => {
+  const name = getFieldValue(vmSettings, VMSettingsField.NAME);
+
+  if (!entity.hasLabel(TEMPLATE_VM_NAME_LABEL)) {
+    entity.addTemplateLabel(TEMPLATE_VM_NAME_LABEL, name);
```

Comment:
it is missing the old comment

Reply:
I would rather not have this one conditional since it is expected to be the name